lib/core/shell.rb

```diff
@@ -55,11 +55,32 @@ def self.verbose_mode(verbose)
     @verbose = verbose
   end
 
-  def self.debug(prefix, message)
-    stderr.puts("\n[#{color(prefix, :red)}] #{message}") if verbose
+  def self.debug(prefix, message, sensitive_data_pattern: nil)
+    filtered_message = hide_sensitive_data(message, sensitive_data_pattern)
```

Comment:
Just a small thing here: since we won't put the message unless `verbose` is `true`, we don't need to execute this line when that happens, so we can just return from the method and remove the condition from below.

Reply:
@ahangarha Please update with check to see if verbose

Reply:
Applied